src/data/nat/enat.lean

```diff
@@ -2,12 +2,48 @@
 Copyright (c) 2018 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes
-
-Natural numbers with infinity, represented as roption ℕ.
 -/
 import data.pfun
 import tactic.norm_num
+import data.equiv.mul_add
+
+/-!
+# Natural numbers with infinity
+
+The natural numbers and an extra `top` element `⊤`.
+
+## Main definitions
+
+The following instances are defined:
+
+* `ordered_add_comm_monoid enat`
+* `canonically_ordered_add_monoid enat`
+
+There is no additive analogue of `monoid_with_zero`; if there were then `enat` could
+be an `add_monoid_with_top`.
+
+* `to_with_top` : the map from `enat` to `with_top ℕ`, with theorems that it plays well
+with `+` and `≤`.
+
+* `with_top_add_equiv : enat ≃+ with_top ℕ`
+* `with_top_order_iso : enat ≃o with_top ℕ`
+
+## Implementation details
+
+`enat` is defined to be `roption ℕ`.
+
+`+` and `≤` are defined on `enat`, but there is an issue with `*` because it's not
+clear what `0 * ⊤` should be. `mul` is hence left undefined.
```

Comment:
```suggestion
## Implementation details

`enat` is defined to be `roption ℕ`.

`+` and `≤` are defined on `enat`, but there is an issue with `*` because it's not
clear what `0 * ⊤` should be. `mul` is hence left undefined. Similarly `⊤ - ⊤` is ambiguous
so there is no `-` defined on `enat`.
```